File: latex/main paper/python code/expX.py
from random import random as U
from math import exp
def X(t): return -t**3*U()**2
def num_B(i):  # = depth of Bernoulli's = 1
    return num_B(i+1) if U()*i < 1 else i-1
def res(n, t): return 1 + X(t)*res(n-1, t) if n != 0 else 1
def expE(t): return res(num_B(0), t)

t, nsim = 1, 10**3
sol = sum(expE(t) for _ in range(nsim))/nsim
exact = exp(-t**3/3)
print(f"sol = {sol} %error={(sol- exact)/exact}")
#sol = 0.7075010309320893 %error=-0.01260277046

File: latex/main paper/julia_code/expX.jl
num_B(i) = rand() * i < 1 ? num_B(i + 1) : i - 1
res(X, n) = (n != 0) ? 1 + X() * res(X, n - 1) : 1
expE(X) = res(X, num_B(0))